ycmd/completers/all/identifier_completer.py

```diff
@@ -37,6 +37,10 @@ def __init__( self, user_options ):
     self._completer = ycm_core.IdentifierCompleter()
     self._tags_file_last_mtime = defaultdict( int )
     self._logger = logging.getLogger( __name__ )
+    try:
+      self._max_candidates = user_options[ 'max_num_identifier_candidates' ]
+    except KeyError:
+      self._max_candidates = MAX_IDENTIFIER_COMPLETIONS_RETURNED
```

Comment:
What do you mean by "config mismatch"? If I don't specify the option then `user_options[ 'max_num_identifier_candidates' ]` will be equal to the default one specified in `default_settings.json`. If I specify the option then `user_options[ 'max_num_identifier_candidates' ]` will be equal to what I specify. Where is the problem I didn't consider?